comms/dht/src/crypt.rs

```diff
@@ -127,10 +125,25 @@ pub fn create_message_challenge_parts(
             buf
         })
         .unwrap_or_default();
-    mac_challenge.update(&e_pk);
 
-    mac_challenge.update(&body);
-    mac_challenge.finalize_fixed().into()
+    // we digest the given data into a domain independent hash function to produce a signature
+    // use of the hashing API for domain separation and deal with variable length input
+    let domain_separated_hash =
+        DomainSeparatedHasher::<Challenge, GenericHashDomain>::new(DOMAIN_SEPARATION_CHALLENGE_LABEL)
+            .chain(&protocol_version.as_bytes())
+            .chain(destination.to_inner_bytes())
+            .chain(&(message_type as i32).to_le_bytes())
+            .chain(&flags.bits().to_le_bytes())
+            .chain(&expires)
+            .chain(&e_pk)
+            .chain(&body)
+            .finalize()
+            .into_vec();
+
+    let mut output = [0u8; 32];
+    // the use of Challenge bind to Blake256, should always produce a 32-byte length output data
+    output.copy_from_slice(domain_separated_hash.as_slice());
+    output
```

Comment:
```suggestion
            .finalize();
        
    domain_separated_hash.hash_to_bytes()
        .expect("Fixed output of '0 < size <= 32' bytes with 32 byte hasher cannot fail.")
```
This way the runtime panic is not hidden in `copy_from_slice`. As an alternative, the function could return a result and the runtime error could be handled.